configure.ac

```diff
@@ -195,6 +180,34 @@ AC_HELP_STRING([--with-sdk-install-path=DIR path to external libs and headers],
 AC_SUBST(SDK_INSTALL_PATH)
 
 ##########################################################################
+# Run platform specific checks and settings
+##########################################################################
+IMPLEMENTATION_NAME=""
+if test "${with_platform}" == "linux-generic";
+then
+    m4_include([./platform/linux-generic/m4/configure.m4])
+    m4_include([./test/linux-generic/m4/configure.m4])
+    IMPLEMENTATION_NAME="linux-generic"
+    ODP_CFLAGS="$ODP_CFLAGS -std=c99"
+    ODP_LIB_STR="odp-linux"
+else
+    if test "${with_platform}" == "linux-dpdk";
+    then
+        m4_include([./platform/linux-dpdk/m4/configure.m4])
+        m4_include([./test/linux-dpdk/m4/configure.m4])
+        IMPLEMENTATION_NAME="linux-dpdk"
+        ODP_CFLAGS="$ODP_CFLAGS -std=gnu99"
+        ODP_LIB_STR="odp-dpdk"
+    else
```

Comment:
No elif, please, it is a non-standard extension. Use AS_IF() macro with multiple test cases.